components/datetime/src/format/datetime.rs

```diff
@@ -532,11 +652,11 @@ where
                 iter.next(); // Advance over nanosecond symbol
                 let r = datetime
                     .nanosecond()
-                    .ok_or(Error::MissingInputField(Some("nanosecond")))
+                    .ok_or(DateTimeWriteError::MissingInputField("nanosecond"))
                     .and_then(|ns| {
                         // We only support fixed field length for fractional seconds.
                         let FieldLength::Fixed(p) = length else {
-                            return Err(Error::Pattern(
+                            return Err(DateTimeWriteError::Pattern(
```

Comment:
Suggestion: This isn't really an _invalid_ pattern; it is an _unsupported_ pattern, I think. Invalid patterns should be handled in Serde. It would be nice to avoid including `PatternError` in the enum. I think it would be okay to return `UnsupportedField` here.

Reply:
I agree. I guess this qualifies as an unsupported field.